### ISSUE TITLE
Horde Flight Master at Dark Portal has green exclamation point

### ISSUE THREAD

**Issue by jerbookins** [TrinityCore/TrinityCore#19398]
[//]: # (**********************************)
[//]: # (** Fill in the following fields **)
[//]: # (**********************************)

Playing as Alliance, the Horde Flight master, Vlagga Freyfeather (ID 18930), has a green exclamation point above her head as if you can learn her flight point but you cannot interact with her.

She shouldn't have an exclamation point above her head.

**Branch(es):**
3.3.5

**TC rev. hash/commit:** 
TrinityCore rev. b7329a460e58 2017-04-01 03:19:57 +0200 (3.3.5 branch) (Win64, Release, Static)

**TDB version:** 
335.62

**Operating system:**
Windows 10 Server & Client

[//]: # (This template is for problem reports. For other types of report, edit it accordingly.)
[//]: # (For fixes containing C++, create a Pull Request.)

**Comment by Kittnz** [comment]
because she has a quest i think

**Comment by jerbookins** [comment]
She has a quest for alliance?

**Comment by Kittnz** [comment]
yeah when you first enter outland she has the quest complete, shes scripted a bit weird i think

**Comment by tkrokli** [comment]
Uhm, AFAIK, the green icon is only to indicate that the current player character hasn't learnt this FP yet.

**Comment by Krudor** [comment]
```
void WorldSession::SendTaxiStatus(ObjectGuid guid)
{
    ...
    else if (unit->GetReactionTo(GetPlayer()) >= REP_NEUTRAL)
        data.Status = GetPlayer()->m_taxi.IsTaximaskNodeKnown(curloc) ? TAXISTATUS_LEARNED : TAXISTATUS_UNLEARNED;
    ...
}

```

Yeah, this doesn't look right...

**Comment by Krudor** [comment]
Dupe of https://github.com/TrinityCore/TrinityCore/issues/18150 ?

**Comment by tkrokli** [comment]
True, good point. Although it does not seem to happen at the same places in `6.x/7.x/master` compared to `3.3.5` branch.

**Comment by MrSmite** [comment]
I think the problem is that some of those Horde NPCs were marked neutral to stop them from attacking the player. Technically should still show the hostile (red) name but just not attack the player (similar to other Horde NPCs like Samantha Swifthoof in Stranglethorn Vale.

[wowhead](http://www.wowhead.com/npc=18930/vlagga-freyfeather) and [gamepedia](http://wow.gamepedia.com/Vlagga_Freyfeather) both show reaction of: Alliance (red) / Horde (green)

**Comment by jerbookins** [comment]
@MrSmite Sounds exactly right

**Comment by MrSmite** [comment]
Hmm... well,

Samantha has *faction=104* (Thunder Bluff) which has *hostileMask=2* (Alliance players and pets) but both the enemy / friend factions are blank.
Samantha has *faction=1760* (Dark Portal Defender Horde) which has *hostileMask=0* but an *enemy faction=993* and *friend faction=992 991*, none of which exist in the [Faction Template table](https://trinitycore.atlassian.net/wiki/display/tc/FactionTemplate)

In fact, all the factions related to "Dark Portal Defender - Horde" have similar fields so I'm not sure how to fix this one...

**Comment by tkrokli** [comment]
Would it be practical to implement those missing factions, or do we need sniffs to prove that they are used on Blizzard servers first (and maybe also how they are used)?

**Comment by Pitcrawler** [comment]
You need to lookup the friend and enemy factions in https://trinitycore.atlassian.net/wiki/display/tc/Faction.

You will see that enemy faction 993 is Dark Portal Attacker, Legion and friend factions are Dark Portal Defender, Alliance and Dark Portal Defender, Horde.

**Comment by Pitcrawler** [comment]
How about unit_flags is UNIT_FLAG_IMMUNE_TO_PC set in sniffs and correct in DB?

**Comment by tkrokli** [comment]
```sql
SELECT * FROM `creature_template` WHERE `entry`= 18930; -- Vlagga Freyfeather
```
```
entry  difficulty_entry_1  difficulty_entry_2  difficulty_entry_3  KillCredit1  KillCredit2  modelid1  modelid2  modelid3  modelid4  name                subname            IconName  gossip_menu_id  minlevel  maxlevel     exp  faction  npcflag  speed_walk  speed_run   scale    rank  dmgschool  BaseAttackTime  RangeAttackTime  BaseVariance  RangeVariance  unit_class  unit_flags  unit_flags2  dynamicflags  family  trainer_type  trainer_spell  trainer_class  trainer_race    type  type_flags  lootid  pickpocketloot  skinloot  resistance1  resistance2  resistance3  resistance4  resistance5  resistance6  spell1  spell2  spell3  spell4  spell5  spell6  spell7  spell8  PetSpellDataId  VehicleId  mingold  maxgold  AIName   MovementType  InhabitType  HoverHeight  HealthModifier  ManaModifier  ArmorModifier  DamageModifier  ExperienceModifier  RacialLeader  movementId  RegenHealth  mechanic_immune_mask  flags_extra  ScriptName  VerifiedBuild
-----  ------------------  ------------------  ------------------  -----------  -----------  --------  --------  --------  --------  ------------------  -----------------  --------  --------------  --------  --------  ------  -------  -------  ----------  ---------  ------  ------  ---------  --------------  ---------------  ------------  -------------  ----------  ----------  -----------  ------------  ------  ------------  -------------  -------------  ------------  ------  ----------  ------  --------------  --------  -----------  -----------  -----------  -----------  -----------  -----------  ------  ------  ------  ------  ------  ------  ------  ------  --------------  ---------  -------  -------  -------  ------------  -----------  -----------  --------------  ------------  -------------  --------------  ------------------  ------------  ----------  -----------  --------------------  -----------  ----------  -------------
18930                   0                   0                   0            0            0     18665         0         0         0  Vlagga Freyfeather  Wind Rider Master  (NULL)              7938        65        65       1     1760     8195           1    1.14286       1       1          0            2000             2000             1              1           1       33536         2048             0       0             0              0              0             0       7   134217728       0               0         0            0            0            0            0            0            0       0       0       0       0       0       0       0       0               0          0        0        0  SmartAI             1            3            1              10            10              1             4.6                   1             0           0            1                     0            2                      12340
```
`unit_flags` = 33536 = 32768 (`UNIT_FLAG_UNK_15`) + 512 (`UNIT_FLAG_IMMUNE_TO_NPC`) + 256 (`UNIT_FLAG_IMMUNE_TO_PC`)
`npcflag` = 8195 = 8192 (Flight Master) + 2 (Quest Giver) + 1 (Gossip)

**Comment by Pitcrawler** [comment]
Well then that npc won't attack any players regardless of its faction as far as I can tell.

**Comment by tkrokli** [comment]
OK, but how to get rid of the FP status showing to the opposite faction (horde vs alliance), is the correct solution to change her faction or do we need to change something in the core to make it work as intended?

**Comment by Aokromes** [comment]
This is how the oposite mob looks on PTR:
http://i.imgur.com/aQW1z8V.png

**Comment by MrSmite** [comment]
> This is how the oposite mob looks on PTR:

But PTR isn't for 3.3.5. Perhaps it would be the easiest way to go since finding 335 info is getting more difficult as time passes.

**Comment by Aokromes** [comment]
I doubt how it works that zone changed since 3.3.5a

**Comment by jerbookins** [comment]
Still broken b978b673a905

**Comment by Killyana** [comment]
This is a core issue related to create/update object, the green exclamation point will be displayed if the npc is created or updated for the client, But if you stand in front of the npc then .gm on .gm off the exclamation point will disappear.

**Comment by cachocarne** [comment]
This is still happening, also from the other faction's point of view. I think it's a minor bug but shouldn't be like that.

![ 2023-05-22 a las 0 05 42](https://github.com/TrinityCore/TrinityCore/assets/77953030/69a0f7c1-5b38-485c-ab0d-5e34f73089d7)
